### PR TITLE
Add Region and Availability zones as attributes to cloudmap instances

### PR DESCRIPTION
**Pull request by achevuru**
*Issue #, if available:*

*Description of changes:*
Add Region and Availability zones as attributes to cloudmap instances

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/cloudmap/resource_manager.go [hunk 1]
```diff
@@ -385,3 +390,32 @@ func (m *defaultResourceManager) isCloudMapServiceOwnedByVirtualNode(ctx context
 func (m *defaultResourceManager) buildCloudMapServiceSummaryCacheKey(nsSummary *servicediscovery.NamespaceSummary, serviceName string) string {
 	return fmt.Sprintf("%s/%s", awssdk.StringValue(nsSummary.Id), serviceName)
 }
+
+func (m *defaultResourceManager) getClusterNodeInfo(ctx context.Context) map[string]nodeAttributes {
```
Comment (fawadkhaliq) [inline pkg/cloudmap/resource_manager.go:394]: Is there a possibility of this going out of sync when a virtualNode is not touched but it's corresponding pod moves to a new node, where the new node is added after this function was called or the reconcile also takes into account Pod updates as well eliminating any possibility of out of sync?

Comment (achevuru) [reply pkg/cloudmap/resource_manager.go:394]: Yeah, CloudMap controller watches for pod events as well. So, we should be good with the above scenario..

File: pkg/cloudmap/instances_reconciler.go [hunk 2]
```diff
@@ -135,19 +140,31 @@ func (r *defaultInstancesReconciler) buildInstanceInfoByID(ms *appmesh.Mesh, vn
...
 	for _, cmAttr := range vn.Spec.ServiceDiscovery.AWSCloudMap.Attributes {
 		attr[cmAttr.Key] = cmAttr.Value
 	}
+	podsNodeName := pod.Spec.NodeName
 	attr[attrAWSInstanceIPV4] = pod.Status.PodIP
 	attr[attrK8sPod] = pod.Name
 	attr[attrK8sNamespace] = pod.Namespace
 	attr[attrAppMeshMesh] = aws.StringValue(ms.Spec.AWSName)
 	attr[attrAppMeshVirtualNode] = aws.StringValue(vn.Spec.AWSName)
+	if nodeInfoByName != nil {
```
Comment (M00nF1sh) [inline pkg/cloudmap/instances_reconciler.go:158]: no need to do nil check for read-only map access :D

File: pkg/cloudmap/resource_manager.go [hunk 3]
```diff
@@ -26,6 +27,9 @@ const (
 	defaultNamespaceCacheTTL               = 2 * time.Minute
 	defaultServiceCacheMaxSize             = 1024
 	defaultServiceCacheTTL                 = 2 * time.Minute
+
+	nodeRegionLabel           = "failure-domain.beta.kubernetes.io/region"
```
Comment (M00nF1sh) [inline pkg/cloudmap/resource_manager.go:31]: let's support "topology.kubernetes.io/region" and "topology.kubernetes.io/zone", these two annotation will replace the beta ones in v1.17 https://github.com/kubernetes/kubernetes/pull/81431

File: pkg/cloudmap/resource_manager.go [hunk 4]
```diff
@@ -385,3 +390,32 @@ func (m *defaultResourceManager) isCloudMapServiceOwnedByVirtualNode(ctx context
...
 }
+
+func (m *defaultResourceManager) getClusterNodeInfo(ctx context.Context) map[string]nodeAttributes {
+	var nodeInfoByName map[string]nodeAttributes
+	nodeList := &corev1.NodeList{}
+	if err := m.k8sClient.List(ctx, nodeList); err != nil {
+		return nodeInfoByName
+	}
+
+	m.log.V(1).Info("Nodes ", "count: ", len(nodeList.Items))
```
Comment (M00nF1sh) [inline pkg/cloudmap/resource_manager.go:401]: better to be 
`m.log.V(1).Info("listed nodes", "count", len(nodeList.Items))`. don't use ":" :D

File: pkg/cloudmap/resource_manager.go [hunk 5]
```diff
@@ -385,3 +390,32 @@ func (m *defaultResourceManager) isCloudMapServiceOwnedByVirtualNode(ctx context
 func (m *defaultResourceManager) buildCloudMapServiceSummaryCacheKey(nsSummary *servicediscovery.NamespaceSummary, serviceName string) string {
 	return fmt.Sprintf("%s/%s", awssdk.StringValue(nsSummary.Id), serviceName)
 }
+
+func (m *defaultResourceManager) getClusterNodeInfo(ctx context.Context) map[string]nodeAttributes {
+	var nodeInfoByName map[string]nodeAttributes
```
Comment (M00nF1sh) [inline pkg/cloudmap/resource_manager.go:395]: don't define nodeInfoByName here if it's not initialized later. we can just return nil if list fails.

**Pull request merged**